daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -48,33 +48,31 @@ private[io] class ByteArrayOutputStreamWithGetBuf() extends java.io.ByteArrayOut
 
 private[io] class ByteArrayOrFileOutputStream(
   maxBufferSizeInBytes: Long,
-  tempFilePath: Path,
-  useFileOnly: Boolean)
-  extends java.io.OutputStream
-  with Logging {
-  val filePath = {
-    if (tempFilePath.toString == "")
-      Paths.get(System.getProperty("java.io.tmpdir"))
-    else
-      tempFilePath
-  }
+  tempDirPath: Path,
+  existingFile: Path)
+  extends java.io.OutputStream with Logging {
 
-  var stream: java.io.OutputStream = {
-    if (useFileOnly)
-      new java.io.FileOutputStream(File.createTempFile("daffodil-", ".tmp", filePath.toFile))
+  val dirPath = {
+    if (tempDirPath.toString == "")
+      Paths.get(System.getProperty("java.io.tmpdir"))
     else
-      new ByteArrayOutputStreamWithGetBuf()
+      tempDirPath
   }
 
-  var isFile: Boolean = useFileOnly
-  var tempFile: File = null
-  var nBytes: Long = 0
+  var isFile: Boolean = (existingFile != null)
```

Comment:
All this null stuff makes me thing existingFile should be a ``Maybe`` so that the logic becomes ``existingFIle.isDefined``. That's more scala-y. And I suspect nothing should really use existingFile. I would expect existingFile would set tempFile (if its defined) and then all the other logic is based on tempFile. So if tempFile is set  then this OutputStream is file-based, otherwise it ByteArray-based.